src/keys/spend/public.rs

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::{
+    permutation, JubJubAffine, JubJubExtended, JubJubScalar, PublicKey,
+    StealthAddress,
+};
+
+use super::secret::SecretSpendKey;
+
+#[cfg(feature = "canon")]
+use canonical::Canon;
+#[cfg(feature = "canon")]
+use canonical_derive::Canon;
+
+use dusk_bytes::{DeserializableSlice, Error, HexDebug, Serializable};
+use dusk_jubjub::GENERATOR_EXTENDED;
+use subtle::{Choice, ConstantTimeEq};
+
+/// Public pair of `a·G` and `b·G` defining a [`PublicSpendKey`]
+#[derive(HexDebug, Clone, Copy)]
+#[cfg_attr(feature = "canon", derive(Canon))]
+pub struct PublicSpendKey {
+    A: JubJubExtended,
+    B: JubJubExtended,
+}
+
+impl PublicSpendKey {
+    /// This method is used to construct a new `PublicSpendKey` from the given
+    /// public pair of `a·G` and `b·G`
+    pub fn new(A: JubJubExtended, B: JubJubExtended) -> Self {
```

Comment:
A public key should never be created out of two isolated points. It should always be derived from a secret key, or from some form of deserialization. Hence, this method should not be public